### ISSUE TITLE
transaction lint triggers for dynamic parameters

### ISSUE THREAD

**Issue by joprice** [Zaid-Ajaj/Npgsql.FSharp.Analyzer#43]
I'm providing the parameters to a transaction query using a list built with a for loop, and this triggers the `Transaction query is parameterized but the parameter sets are empty` lint. Inlining static values silences it. Is there a way to write dynamic transaction params to avoid the false positive?

**Comment by Zaid-Ajaj** [comment]
I believe the analyzer isn't well equipped to handle setting parameters from loops, probably not easy to implement but happy to help if you are willing to take a look at it

**Comment by joprice** [comment]
Yea it's probably a bit tricky, as there's so many forms that could result in a valid set of parameters, but perhaps it just traverses the tree recursively looking for at least one call to the parameters, even if there are false positives that would never set the parameter like perhaps

```fsharp
[
  if false then 
    yield "x", Sql.string "a"
  yield "y", Sql.int 1
]
```

A best effort attempt to collect all parameter tuples and checking the set for at least the union to equal the expected might be enough.
